dali/image/tiff.cc

```diff
@@ -0,0 +1,111 @@
+#include "tiff.h"
+
+namespace dali {
+
+namespace {
+
+const std::vector<int> header_intel = {77, 77, 0, 42};
+const std::vector<int> header_motorola = {73, 73, 42, 0};
+const int COUNT_SIZE = 2;
+const int ENTRY_SIZE = 12;
+const int WIDTH_TAG = 256;
+const int HEIGHT_TAG = 257;
+
+
+cv::Mat DecodeTiff(const unsigned char *tiff, int size, DALIImageType image_type) {
+    std::vector<char> buf(tiff, tiff + size);
+    return cv::imdecode(buf, image_type == DALI_GRAY ? 0 : 1);
+}
+
+
+bool check_header(const unsigned char *tiff, const std::vector<int> &header) {
+    for (unsigned int i = 0; i < header.size(); i++) {
+        if (tiff[i] != header[i]) {
+            return false;
+        }
+    }
+    return true;
+}
+
+
+bool is_little_endian(const unsigned char *tiff) {
+    return check_header(tiff, header_intel);
+}
+
+} // namespace
+
+bool CheckIsTiff(const unsigned char *tiff) {
+    assert(tiff);
+    return check_header(tiff, header_intel) || check_header(tiff, header_motorola);
+}
+
+
+DALIError_t GetTiffImageDims(const unsigned char *tiff, int size, int *h, int *w) {
+    assert(CheckIsTiff(tiff));
+
+    tiff_buffer buffer(std::string(reinterpret_cast<const char *>(tiff), static_cast<size_t>(size)),
+                       is_little_endian(tiff));
+
+    auto ifd_offset = buffer.read<unsigned int>(4);
+    auto entry_count = buffer.read<unsigned short>(ifd_offset);
+    bool width_read = false, height_read = false;
+
+    for (int entry_idx = 0;
+         entry_idx < entry_count && !(width_read && height_read);
+         entry_idx++) {
+        auto entry_offset = ifd_offset + COUNT_SIZE + entry_idx * ENTRY_SIZE;
+        auto tag_id = buffer.read<unsigned short>(entry_offset);
+        if (tag_id == WIDTH_TAG || tag_id == HEIGHT_TAG) {
+            auto value_type = buffer.read<unsigned short>(entry_offset + 2);
+            auto value_count = buffer.read<unsigned int>(entry_offset + 4);
+            assert(value_count == 1);
+
+            unsigned long value;
+            if (value_type == 3) {
```

Comment:
You can use some define for those 3 and 4 values.

Reply:
OK